storage_controller/src/reconciler.rs

```diff
@@ -848,14 +848,19 @@ impl Reconciler {
     }
 
     /// Keep trying to notify the compute indefinitely, only dropping out if:
-    /// - the node `origin` becomes unavailable
-    /// - the node `origin` no longer has our tenant shard attached
-    /// - our cancellation token fires
+    /// - the node `origin` becomes unavailable -> Ok(())
+    /// - the node `origin` no longer has our tenant shard attached -> Ok(())
+    /// - our cancellation token fires -> Err(ReconcileError::Cancelled)
     ///
     /// This is used during live migration, where we do not wish to detach
     /// an origin location until the compute definitely knows about the new
     /// location.
-    async fn compute_notify_blocking(&mut self, _origin: &Node) -> Result<(), ReconcileError> {
+    ///
+    /// In cases where the origin node becomes unavailable, we return success, indicating
+    /// to the caller that they should continue irrespective of whether the compute was notified,
+    /// because the origin node is unusable anyway.  Notification will be retried later via the
+    /// [`Self::compute_notify_failure`] flag.
+    async fn compute_notify_blocking(&mut self, origin: &Node) -> Result<(), ReconcileError> {
```

Comment:
Not necessarily related to your PR, but what about the node getting dropped mid migration?
Should we explicitly check that the node is still registered below?

Reply:
We should make sure that when the node is dropped its cancellation token is fired.

Added that explicitly in node_delete in bdda8a797